### DISCUSSION TITLE
Interaction of Table include_columns and extend_existing can remove columns

### DISCUSSION THREAD

**Discussion by CaselIT** [sqlalchemy/sqlalchemy#6948] [Usage Questions]
Table currently has no public method to remove columns, but I noticed that this can be achieved by carefully setting `include_columns` and `extend_existing` .
Example:
```py
import sqlalchemy as sa
meta = sa.MetaData()
t = sa.Table('tt', meta, sa.Column('a', sa.Integer()))
print(repr(t))  # will print Table('tt', MetaData(), Column('a', Integer(), table=<tt>), schema=None)

sa.Table('tt', meta, include_columns=[], extend_existing=True)
print(repr(t)) # will print Table('tt', MetaData(), schema=None)
```

Is this something that should be officially supported? I think the current behaviour is an accidental one.
My use case is to clear the column of a dynamically generated view, that's reflected as a table, during tests.

**Comment by zzzeek** [comment]
that worked by accident.   removal of columns, ORM attributes, etc., none of that is supported.  removal is *way* more complicated than addition , just see the events package , where i had to make the tough decision *to* support removal, for what a hard problem that is.  all that weakref stuff, etc.

**Reply by CaselIT** [reply]
That was my guess as well. It works for my use case in the tests, but seems quite by accident.

Do you have other suggestions regarding thins. Is there a way to "reset" a table (and should there be?)

**Reply by zzzeek** [reply]
IMO there shouldnt be a table "reset".   if something changes on every run then youd use a new MetaData/ Table each time, or even if it's just a view use the inspector and create a quick table(..., column()...) yourself.

**Reply by CaselIT** [reply]
Ok.

Should the above condition flagged with a warning in sqlalchemy?

>  if something changes on every run then youd use a new MetaData/ Table each time

yes that's an alternative. it does not really changes each test, more like while changing test package

**Reply by zzzeek** [reply]
it depends on how hard / arbitrary etc it is.
